server/src/test/java/org/elasticsearch/rest/action/RestCancellableNodeClientTests.java

```diff
@@ -303,6 +303,10 @@ public void addCloseListener(ActionListener<Void> listener) {
             // if the channel is already closed, the listener gets notified immediately, from the same thread.
             if (open.get() == false) {
                 listener.onResponse(null);
+                // Handle scenario where awaitClose() was called before any calls to addCloseListener(), this ensures closeLatch is pulled.
```

Comment:
Ah good point, but then could we achieve the same thing by dropping the whole `if (open.get() == false)` and always calling `closeListener.onResponse(ActionListener.assertOnce(listener));`? I guess that doesn't guarantee to complete `listener` on the calling thread in that case, not sure if this is important.